rangeserver/src/server.rs

```diff
@@ -46,40 +46,45 @@ impl RangeServer for ProtoServer {
     }
 }
 
-pub struct Server<S, E>
+pub struct Server<S, E, C>
 where
     S: Storage,
     E: EpochProvider,
+    C: Cache,
 {
     config: Config,
     storage: Arc<S>,
     epoch_provider: Arc<E>,
+    cache: Arc<RwLock<C>>,
```

Comment:
I think there should be one range cache per range, and not a global one. As discussed, they could still all use the same underlying rocksdb instance etc.

Reply:
Agree. I will fix this.